### PR TITLE
Made query to append file and document data to the asset

### PR DESCRIPTION
**Pull request by lperezdude**
ADdde quesry to get gallery data

### CODE REVIEW

**Comment by lperezdude** [comment]
Added manipulators to the assets package, this will be the final addition to the Assets service,
the assets package will now continue development here[https://github.com/Combustion-Group/assets](url)

File: src/Services/Assets/DocumentsGateway.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+     */
+    public function getManipulator(array $options=[]) : Manipulator
+    {
+        // if model is not sent return default manipulator
+        if(!isset($options['model']))return $this->manipulators[$this->config['default_manipulator']];
+        // if it was sent make sure it has the HasAssets trait otherwise throw exception
+        if(!isset(class_uses($options['model'])[HasAssets::class])) throw new ModelMustHaveHasAssetsTrait(get_class($options['model'])." does not have HasAssets trait");
+        // if the method does exist return getManipulator from model
+        if(method_exists($options['model'],'getManipulator')) return $this->manipulators[$options['model']->getManipulator()];
+        // otherwise return default manipulator again
```
Comment (the-invisible-man) [inline src/Services/Assets/DocumentsGateway.php]: Make these a little more clear to read.
```
if (condition) {
    // some execution
} elseif (condition) {
     // some execution
}
```

File: src/Services/Assets/Manipulators/BannerImageManipulator.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        $data=[
+            'width'=>isset($options['width'])?$options['width']:0,
+            'height'=>isset($options['height'])?$options['height']:0,
+            'x'=>isset($options['x'])?$options['x']:0,
+            'y'=>isset($options['y'])?$options['y']:0,
+        ];
+        foreach ($data as $coordinates=>$value) if($value===0) throw new ImageDimensionsAreInvalid(ucfirst($coordinates)." cannot be empty or have a value of 0");
+        if((int)$data['width']*.5625!=(int)$data['height']) throw new InvalidAspectRatio("Height adn Width given are not 16:9 aspect ratio");
+        return $data;
+    }
```
Comment (the-invisible-man) [inline src/Services/Assets/Manipulators/BannerImageManipulator.php]: Make a little more readable.

File: src/Services/Assets/Traits/BaseModel.php [hunk 3]
```diff
@@ -0,0 +1,688 @@
...
+     *
+     * @param null $transformerName
+     * @return mixed
+     */
+    public function CustomTransform($transformerName = null)
+    {
+        // get current array
+        $item = $this->convertToArray();
+        // get correct transformer
+        $transformer = $this->getTransformer($transformerName);
```
Comment (the-invisible-man) [inline src/Services/Assets/Traits/BaseModel.php]: This `getTransformer` method returns the same value that you're feeding. And if no value is passed to the function it'll just return null.

File: src/Services/Assets/Traits/BaseModel.php [hunk 4]
```diff
@@ -0,0 +1,688 @@
+<?php
+namespace Combustion\StandardLib\Services\Assets\Traits;
```
Comment (the-invisible-man) [inline src/Services/Assets/Traits/BaseModel.php:2]: There's a good amount of type hints missing in the methods in this file.

**Pull request merged**